unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Video/DCLVideoTexture.cs

```diff
@@ -187,9 +199,32 @@ private IEnumerator VideoTextureUpdate()
                     texturePlayer.UpdateWebVideoTexture();
                 }
 
+                if (texturePlayer.playing && IsTimeToReportVideoProgress())
+                {
+                    lastVideoProgressReportTime = Time.unscaledTime;
+                    ReportVideoProgress();
+                }
+
                 yield return null;
             }
         }
+        private void ReportVideoProgress()
+        {
+            var videoStatus = GetVideoStatus();
+            var currentOffset = texturePlayer.GetTime();
+            var length = texturePlayer.GetDuration();
+            WebInterface.ReportVideoProgressEvent(id, scene.sceneData.id, lastVideoClipID, videoStatus, currentOffset, length );
+        }
+        private int GetVideoStatus()
+        {
+            //TODO: Handle buffering state that returns 2
+
+            if (texturePlayer.playing)
+                return 1;
```

Comment:
Same here, this number may better be a constant or enum somewhere to prevent accidents and to make the code easier to read and maintain

Reply:
Hmm this message structure is a DTO, I'm not sure if it's ok to create an enum for this because we already have a VideoState enum that has its own status
```c#
    public enum VideoState
    {
        NONE = 0,
        ERROR = 1,
        LOADING = 2,
        READY = 3,
        PLAYING = 4,
        BUFFERING = 5
    };
```
Maybe we can use the same status numbers at sdk.